### PR TITLE
prometheus: update to 2.29.1

### PR DESCRIPTION
**Pull request by aparcar**
Signed-off-by: Paul Spooren <mail@aparcar.org>

### CODE REVIEW

**Comment by paper42** [comment]
This PR doesn't really work for me, I am running prometheus from CLI and accessing localhost:9090 through an ssh tunnel. Going to `http://localhost:9090/graph` returns code 500 with `Error opening React index.html: open static/react/index.html: no such file or directory`.

Installing /prometheus.yml sounds wrong.
```
# opkg files prometheus
Package prometheus (2.29.1-2) is installed on root and has the following files:
/prometheus.yml
/etc/init.d/prometheus
/usr/bin/prometheus
/etc/prometheus.yml
/etc/uci-defaults/prometheus-defaults
```

some files installed in prometheus-web-console also sound wrong - index.html.example?
```
# opkg files prometheus-web-console
Package prometheus-web-console (2.29.1-2) is installed on root and has the following files:
/usr/share/prometheus/consoles/index.html.example
/usr/share/prometheus/consoles/node-disk.html
/usr/share/prometheus/console_libraries/prom.lib
/usr/share/prometheus/consoles/prometheus.html
/usr/share/prometheus/consoles/node-overview.html
/usr/share/prometheus/consoles/node.html
/usr/share/prometheus/consoles/node-cpu.html
/usr/share/prometheus/consoles/prometheus-overview.html
/usr/share/prometheus/console_libraries/menu.lib
```

File: utils/prometheus/Makefile [hunk 1]
```diff
@@ -42,22 +42,42 @@ $(call Package/prometheus/Default)
...
+
 define Package/prometheus/install
 	$(call GoPackage/Package/Install/Bin,$(1))
 
 	$(CP) ./files/* $(1)/
 endef
 
+define Package/prometheus-web-console/install
+	$(INSTALL_DIR) $(1)/usr/share/prometheus/consoles
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/consoles/* $(1)/usr/share/prometheus/consoles/
```
Comment (paper42) [inline utils/prometheus/Makefile:71]: consoles/ contains .html files, they don't have to have 755 permissions

**Comment by aparcar** [comment]
@paper42 do you think there is an extra compile step involved? I looked at how Prometheus creates the package itself and couldn't find any special modifications of the web console stuff...

File: utils/prometheus/Makefile (hunk 1 above)
Comment (aparcar) [reply utils/prometheus/Makefile:71]: fair, this could be INSTALL_DATA

**Comment by paper42** [comment]
> @paper42 do you think there is an extra compile step involved? I looked at how Prometheus creates the package itself and couldn't find any special modifications of the web console stuff...

I am not sure, but in Void Linux we run make assets which runs yarn to generate the web console:
```
CGO_ENABLED=0 GOARCH= make assets
```
source: https://github.com/void-linux/void-packages/blob/master/srcpkgs/prometheus/template#L27-L29

**Comment by aparcar** [comment]
Huh, it wants yarn to be installed? I may switch to release builds to avoid such dependencies...

**Comment by aparcar** [comment]
Outdated.

**Pull request closed without merging**